eos/utils/integrate.cc

```diff
@@ -20,14 +20,27 @@
 #include <eos/utils/integrate.hh>
 #include <eos/utils/matrix.hh>
 
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_integration.h>
+
 #include <limits>
 #include <vector>
 
+namespace
+{
+    double gsl_function_adapter(double x, void *params)
+    {
+        const auto& f = *static_cast<eos::GSL::fdd*>(params);
```

Comment:
Are you using eos::GSL::fdd anywhere else? How about:
``` C++
template <typename Result_, typename ... Arguments_>
Result_ std_function_adapter(Arguments_ ... args, void * params)
{
    const auto & f = *static_cast<std::function<Result_ (Arguments_ ...)> *>(params);

    return f(args ...);
}
```

Completely untested!

Reply:
No, but I think we should have this type. But it doesn't have to be in `GSL` because it's more widely used. It hurts to see the same type `std::function<double (const double &)> ` repeated all over the place, e.g. in `integrate.hh`, `integrate_TEST.cpp` etc.

Reply:
I agree, the repetition is not nice.